uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js

```diff
@@ -728,6 +757,22 @@ define([
       }
 
       /**
+       * Changes status of the leave request before saving it
+       */
+      function changeStatusBeforeSave() {
+        if (this.isRole('staff')) {
+          // for staff the status_id should be always set to awaitingApproval before saving
```

Comment:
put this explanations in the function documentation, you are describing the "business logic" of the function